blocks/main_loop.py

```diff
@@ -231,8 +231,19 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                if other not in ['before_batch', 'after_batch',
+                                 'on_resumption', 'on_error',
+                                 'before_training', 'before_epoch',
+                                 'after_epoch', 'after_training',
+                                 'on_interrupt']:
+                    raise ValueError("I'm being compared to \
```

Comment:
Thanks for your detailed comments. I made some changes to the code and push it again. As you see, it has the same mistake as before. Did I do sth wrong with it? Sorry for keeping asking so many questions. :)